Add all core testing infrastructure

taywrobel opened the pull request:
Adds all core automation and some testing harnesses.

taywrobel committed:
Added Automation Package and Smoke Test

Added the robotium package to the testing
package and a working smoke test, that opens
every app and asserts that it was opened
correctly.

taywrobel committed:
Added first harness support

Added basic testing harnesses for the connect
and menu fragments.  Support will grow over
time as tests are written.

Also improved upon the smoke test, making it
much more succinct.

taywrobel committed:
Documentation of automation classes

Also, warning free code is happy code.
Removed unused imports and unclosed streams
out the wazoo.

taywrobel committed:
Added Spoon Library Support to Testing

Can now spoon all our devices!

taywrobel committed:
Added check for welcome dialog in base test

And commited updated test classpath file, which
references robotium and spoon.

taywrobel committed:
Built up additional harness support

Added basic harness support for the
About and menu fragments, as well as
wrote some more simple tests, and cleaned
up the smoke test.

dgreenhalgh commented:
I'll read through all of this later, but I love the sound of it

dgreenhalgh commented:
Could you add a setup description for the dependencies in our docs folder?

taywrobel committed:
Update dependencies file with test dependencies

Added instructions for setting up test dependencies
to the dependencies.md file.

reidbaker commented on SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/ConnectHarness.java:
```diff
@@ -0,0 +1,121 @@
+	}
+
+	public void answerBluetoothDialog(boolean respondYes) {
+		// FIXME: Doesn't work
```
Is this fix me supposed to happen before it gets pulled in? If not is there a ticket to fix this?

reidbaker commented on SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/ConnectHarness.java:
```diff
@@ -0,0 +1,121 @@
+		Condition con = new Condition() {
+			@Override
+			public boolean isSatisfied() {
+				// FIXME: Doesn't work
```
same as above

reidbaker commented on SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/SoundStreamHarness.java:
```diff
@@ -0,0 +1,76 @@
+	}
+
+	public void assertTitleEquals(String expectedTitle) {
+		// FIXME: Doesn't grab title as it should. Should pull from UI directly.
```
same here

reidbaker commented on SoundStream-test/src/com/lastcrusade/soundstream/ete/suites/AllEteTests.java:
```diff
@@ -0,0 +1,46 @@
+ * @author Taylor Wrobel
+ * 
+ */
+public class AllEteTests extends TestCase {
```
Looks funny... is ete an abbreviation? Should we change this to AllETETests?

taywrobel replied:
I would go with no.  This was a huge debate at work for a while, but we decided to only camel case the starts of acronyms.  In this case Ete is End-to-end.

dgreenhalgh replied:
I've seen it go both ways - for example I've seen the shorthand for identification as "ID" and "Id"

reidbaker commented on SoundStream-test/src/com/lastcrusade/soundstream/util/EteUtils.java:
```diff
@@ -0,0 +1,55 @@
+		
+		boolean vis = false;
+		
+		for(Fragment f : core.getAttachedFragments()){
```
I have no idea what this block of code does. Could we get some comments into this :poop:

reidbaker commented on SoundStream/res/layout/fragment_connect.xml:
```diff
@@ -61,6 +63,7 @@
         
         <ImageView
             android:contentDescription="@string/join_network"
+	        android:id="@+id/join_Img"
```
Spacing looks funny here

reidbaker commented:
I have a few comment but otherwise this looks great

taywrobel committed:
Adds comments to the isFragmentVisible method

Describes in more detail the logic behind determining
if a fragment is visible, for testing pursposes.
